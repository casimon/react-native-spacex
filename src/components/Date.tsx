import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  StyleProp,
  TextStyle,
  ViewStyle,
} from 'react-native';

import moment from 'moment';

const styles = StyleSheet.create({
  date: {
    flex: 1,
  },
  dateText: {
    fontSize: 12,
    color: '#333333',
  },
});

interface Props {
  date: string;
  format: string;
  style?: StyleProp<ViewStyle>;
  styleText?: StyleProp<TextStyle>;
}

class Date extends React.Component<Props> {
  formatDateToLocal = (date: string, format: string) => {
    return moment
      .utc(date)
      .local()
      .format(format);
  };

  render() {
    const {date, format} = this.props;
    return (
      <View style={[styles.date, this.props.style]}>
        <Text style={[styles.dateText, this.props.styleText]}>
          {this.formatDateToLocal(date, format)}
        </Text>
      </View>
    );
  }
}

export default Date;
