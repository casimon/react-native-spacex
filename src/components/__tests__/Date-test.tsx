/**
 * @format
 */

import 'react-native';
import React from 'react';
import Date from '../Date';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders date correctly with date string', () => {
  const tree = renderer
    .create(<Date date="2019-05-04T06:48:00.000Z" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders date correctly with date string and optional style', () => {
  const tree = renderer
    .create(<Date date="2019-05-04T06:48:00.000Z" style={{margin: 2}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders date correctly with date string and optional styleText', () => {
  const tree = renderer
    .create(<Date date="2019-05-04T06:48:00.000Z" styleText={{fontSize: 14}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
