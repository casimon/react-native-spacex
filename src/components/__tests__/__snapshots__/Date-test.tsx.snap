// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders date correctly with date string 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 3,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 12,
        },
        undefined,
      ]
    }
  >
    Sat, May 4, 2019 4:48 PM
  </Text>
</View>
`;

exports[`renders date correctly with date string and optional style 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 3,
      },
      Object {
        "margin": 2,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 12,
        },
        undefined,
      ]
    }
  >
    Sat, May 4, 2019 4:48 PM
  </Text>
</View>
`;

exports[`renders date correctly with date string and optional styleText 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 3,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 12,
        },
        Object {
          "fontSize": 14,
        },
      ]
    }
  >
    Sat, May 4, 2019 4:48 PM
  </Text>
</View>
`;
